backend/app/adapter/gqlapi/resolver/authmutation_test.go

```diff
@@ -0,0 +1,269 @@
+			relationShortLinks: []entity.ShortLink{
+				{
+					Alias:    "SimpleAlias",
+					LongLink: "https://www.google.com/",
+				},
+			},
+			expectedShortLink: nil,
+			shouldError:       false,
+		},
+		{
+			name: "update only alias",
+			args: &UpdateURLArgs{
+				OldAlias: "SimpleAlias",
+				URL: URLInput{
+					CustomAlias: &newAlias,
+				},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "short@gmail.com",
+			},
+			urls: urls,
+			relationUsers: []entity.User{
+				{
+					ID:    "1",
+					Email: "short@gmail.com",
+				},
+			},
+			relationShortLinks: []entity.ShortLink{
+				{
+					Alias:    "SimpleAlias",
+					LongLink: "https://www.google.com/",
+				},
+			},
+			expectedShortLink: &URL{
+				url: entity.ShortLink{
+					Alias:    newAlias,
+					LongLink: "https://www.google.com/",
+				},
+			},
+			shouldError: false,
+		},
+		{
+			name: "update only long link",
+			args: &UpdateURLArgs{
+				OldAlias: "SimpleAlias",
+				URL: URLInput{
+					OriginalURL: &newLongLink,
+				},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "short@gmail.com",
+			},
+			urls: urls,
+			relationUsers: []entity.User{
+				{
+					ID:    "1",
+					Email: "short@gmail.com",
+				},
+			},
+			relationShortLinks: []entity.ShortLink{
+				{
+					Alias:    "SimpleAlias",
+					LongLink: "https://www.google.com/",
+				},
+			},
+			expectedShortLink: &URL{
+				url: entity.ShortLink{
+					Alias:    "SimpleAlias",
+					LongLink: newLongLink,
+				},
+			},
+			shouldError: false,
+		},
+		{
+			name: "update both alias and long link",
+			args: &UpdateURLArgs{
+				OldAlias: "SimpleAlias",
+				URL: URLInput{
+					CustomAlias: &newAlias,
+					OriginalURL: &newLongLink,
+				},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "short@gmail.com",
+			},
+			urls: urls,
+			relationUsers: []entity.User{
+				{
+					ID:    "1",
+					Email: "short@gmail.com",
+				},
+			},
+			relationShortLinks: []entity.ShortLink{
+				{
+					Alias:    "SimpleAlias",
+					LongLink: "https://www.google.com/",
+				},
+			},
+			expectedShortLink: &URL{
+				url: entity.ShortLink{
+					Alias:    newAlias,
+					LongLink: newLongLink,
+				},
+			},
+			shouldError: false,
+		},
+		{
+			name: "update long link to malicious url",
+			args: &UpdateURLArgs{
+				OldAlias: "SimpleAlias",
+				URL: URLInput{
+					OriginalURL: &maliciousURL,
+				},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "short@gmail.com",
+			},
+			urls: urls,
+			relationUsers: []entity.User{
+				{
+					ID:    "1",
+					Email: "short@gmail.com",
+				},
+			},
+			relationShortLinks: []entity.ShortLink{
+				{
+					Alias:    "SimpleAlias",
+					LongLink: "https://www.google.com/",
+				},
+			},
+			shouldError: true,
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+			blockedHash := map[string]bool{
+				"http://malware.wicar.org/data/ms14_064_ole_not_xp.html": false,
```

Comment:
```suggestion
				maliciousURL: false,
```

Reply:
This will not work for the blacklist since it requires a string for the key and a boolean for the value. The url is used as the key so that a for loop is not needed when checking if a url exists within the blacklist. The underlying implementation for this uses something similar to the following:
```Golang
if _, ok := blacklist[url]; !ok {
```